next/test/visibility.test.ts

```diff
@@ -1,271 +1,267 @@
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(false)
+
+        // Different name provided
+        form.handleValidation({
+          name: 'some name',
+          password: null,
+        })
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(false)
+      })
+
+      it('should show the password field if the name is admin', () => {
+        const form = createHeadlessForm(schema)
+        form.handleValidation({
+          name: 'admin',
+        })
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(true)
+      })
     })
-
-    it('should hide fields explicitly set to false in properties', () => {
+    describe('if an "else" branch is not provided', () => {
+      const userName = 'user that does not need password field visible'
       const schema: JsfObjectSchema = {
         type: 'object',
         properties: {
-          toggle: { type: 'boolean' },
-          alwaysVisible: { type: 'string' },
+          name: {
+            type: 'string',
+          },
+          password: {
+            type: 'string',
+          },
         },
-        required: ['alwaysVisible'],
         allOf: [
           {
             if: {
               properties: {
-                toggle: { const: true },
+                name: {
+                  const: userName,
+                },
               },
-              required: ['toggle'],
+              required: ['name'],
             },
             then: {
               properties: {
-                alwaysVisible: false,
+                password: false,
               },
             },
           },
         ],
       }
 
-      const form = createHeadlessForm(schema)
-
-      const targetField = form.fields.find(field => field.name === 'alwaysVisible')
-      expect(targetField?.isVisible).toBe(true)
-
-      const value: ObjectValue = { toggle: true as unknown as SchemaValue }
-      form.handleValidation(value)
-      expect(targetField?.isVisible).toBe(false)
+      it('should show the password field by default', () => {
+        const form = createHeadlessForm(schema)
+        // No name provided
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(true)
+
+        // Different name provided
+        form.handleValidation({
+          name: 'some name',
+        })
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(true)
+      })
+
+      it('should hide the password field if the name is "user that does not need password field visible"', () => {
+        const form = createHeadlessForm(schema, { initialValues: { name: userName, password: null } })
+        expect(form.fields.find(field => field.name === 'password')?.isVisible).toBe(false)
+      })
     })
-
-    it('should handle nested field visibility', () => {
+    describe('if no "else" or "then" branch are provided', () => {
```

Comment:
q: What's the purpose of this test? Seems redundant/useless having a if without then/else.

Reply:
Just being extra cautious